### ISSUE TITLE
Multiple `filter` for `guard`

### ISSUE THREAD

**Issue by Kelin2025** [effector/effector#551]
## Proposal
If `filter` is an array of stores, combine them into one with false unless **all** of stores are true

## Use case
I have canvas editor with draggable blocks + surface, and I need to track block dragging only when drag was started inside block AND canvas is not currently dragging
This notation would be cool:
```tsx
guard({
  clock: panChanged,
  filter: [$isSomeBlockDragging, $isCanvasNotDragging]
})
```
But now it's not supported and we need to use extra combine here
```tsx
guard({
  clock: panChanged,
  filter: combine($isSomeBlockDragging, $isCanvasNotDragging, (...checks) => checks.every(Boolean))
})
```

**Comment by igorkamyshev** [comment]
Look at [patronum/some](https://github.com/effector/patronum#some) and [patronum/every](https://github.com/effector/patronum#every).

In my opinion, it's better to define strategy explicitly.

**Comment by zerobias** [comment]
I suggest to go deeper

```ts
guard({
  clock,
  filter: {
    mode: 'every',
    units: [$flagA, $flagB],
  },
  target,
})

guard({
  clock,
  filter: {
    mode: 'some',
    units: [$flagA, $flagB],
  },
  target,
})

```

**Comment by igorkamyshev** [comment]
Are you serious?

**Comment by Kelin2025** [comment]
And `none` as well 🌚

**Comment by Kelin2025** [comment]
Actually, this idea came to my head because we do the same with `source`
```tsx
sample({ source: [a, b, c] }) === sample({ source: combine([a, b, c]) })
```
And having the same thing for `filter` doesn't sound that bad 🤔

**Comment by zerobias** [comment]
Is not obvious why `'every'` use case is better to be default than `'some'`

**Comment by doasync** [comment]
> why 'every' use case is better to be default than 'some'

Maybe because the array in the filter is intuitively implied as a set of preconditions, and a precondition must always be true just before an operation.

**Comment by zerobias** [comment]
> intuitively implied as a set of preconditions

Indeed. "Set of something" don't imply how that set will be used. For example, array in `clock` is obviously an "or" clause ("some of that units are triggered"), example of "and" clause is `combineEvents` from patronum

**Comment by zerobias** [comment]
I’ve had time to think about this proposal.

First, the case described by the author ("all") is much more important than the one I suggested ("some"), that is, we can focus at the original api.
Second, there are a number of cases where we need to do filtering through both units and functions. That is, it is possible to extend the proposal like this:

```ts
guard({
  clock: panChanged,
  filter: [
    $isSomeBlockDragging,
    $isCanvasNotDragging,
    (pan) => pan.touches.length > 0,
  ]
})
```
